### PR TITLE
Convert to androidx & update the SDK

### PR DESCRIPTION
**Pull request by C2H6O**

### CODE REVIEW

**Comment by benasher44** [comment]
Verified that this builds locally and tested it with the android app

File: gradle.properties [hunk 1]
```diff
@@ -9,9 +9,14 @@
...
 # The setting is particularly useful for tweaking memory settings.
+android.enableJetifier=true
+android.useAndroidX=true
 org.gradle.jvmargs=-Xmx1536m
 
-# When configured, Gradle will run in incubating parallel mode.
-# This option should only be used with decoupled projects. More details, visit
-# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-# org.gradle.parallel=true
+ANDROID_MIN_SDK_VERSION=21
```
Comment (diegoescudero) [inline gradle.properties]: Lets define these outside of the gradle.properties file

File: sqlite-fts5/build.gradle [hunk 2]
```diff
@@ -48,12 +44,12 @@ android {
             path 'src/main/CMakeLists.txt'
         }
     }
-    buildToolsVersion '26.0.2'
+    buildToolsVersion ANDROID_BUILD_TOOLS_VERSION
```
Comment (diegoescudero) [inline sqlite-fts5/build.gradle]: missed toInteger()?

File: app/build.gradle [hunk 3]
```diff
@@ -1,35 +1,36 @@
...
     }
 }
 
 dependencies {
-    implementation 'com.android.support:appcompat-v7:26.1.0'
-    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
-    implementation 'io.requery:sqlite-android:3.22.0'
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.0-alpha4'
+    implementation 'androidx.sqlite:sqlite:2.0.1'
```
Comment (diegoescudero) [inline app/build.gradle:29]: Why do we need this in addition to requery?

**Review by diegoescudero — CHANGES_REQUESTED** [review]

File: sqlite-fts5/build.gradle (hunk 2 above)
Comment (benasher44) [reply sqlite-fts5/build.gradle]: this one actually expects a string, but if I get the ext stuff work, then I think I can remove the `.toInteger` stuff

File: app/build.gradle (hunk 3 above)
Comment (benasher44) [reply app/build.gradle:29]: This is so this library can see the androidx sqlite support classes (won't build without it)

**Comment by benasher44** [comment]
@diegoescudero okay this is ready for (hopefully) a final round! just ran against plangrid unit tests, and things are looking good

**Review by diegoescudero — APPROVED** [review]

**Pull request merged**